packages/drivers/odsp-driver/src/compactSnapshotParser.ts

```diff
@@ -17,69 +19,23 @@ import {
 
-/**
- *  Header of the downloaded snapshot.
- */
-interface ISnapshotHeader {
-    // This is the minimum version of the reader required to read the wire format of a given snapshot.
-    MinReadVersion: string,
-    // Represents the version with which snapshot is created.
-    CreateVersion: string,
-    // Seq number at which the snapshot is created.
-    SnapshotSequenceNumber: number,
-    // Id of the snapshot.
-    SnapshotId: string,
-}
-
-/**
- * Recreates header section and then validate that.
- * @param node - tree node to read header section from
- */
-function readAndValidateHeaderSection(node: NodeCore): ISnapshotHeader {
-    const records =
-        getAndValidateNodeProps(node,
-            ["MinReadVersion", "CreateVersion", "SnapshotSequenceNumber", "SnapshotId"]);
-
-    const header: ISnapshotHeader = {
-        MinReadVersion: records.MinReadVersion.toString(),
-        CreateVersion: records.CreateVersion.toString(),
-        SnapshotSequenceNumber: records.SnapshotSequenceNumber.valueOf() as number,
-        SnapshotId: records.SnapshotId.toString(),
-    };
-    assert(snapshotMinReadVersion >= header.MinReadVersion,
-        0x20f /* "Driver min read version should >= to server minReadVersion" */);
-    assert(header.CreateVersion >= snapshotMinReadVersion,
-        0x210 /* "Snapshot should be created with minReadVersion or above" */);
-    return header;
-}
-
-/**
- * Recreates dictionary section of the tree.
- * @param node - tree node to read dictionary section from. This container paths/ids mapping
- * to integer representation.
- */
-function readDictionarySection(node: NodeCore) {
-    const dictionary = new Array<string>();
-    for (const childNode of node) {
-        assertBlobCoreInstance(childNode, "Mapping should be of type BlobCore");
-        dictionary.push(childNode.toString());
-    }
-    return dictionary;
+interface ISnapshotSection {
+    snapshotTree: ISnapshotTree,
+    sequenceNumber: number,
 }
 
 /**
  * Recreates blobs section of the tree.
  * @param node - tree node to read blob section from
- * @param dictionary - name map, used to decode path/IDs.
  */
-function readBlobSection(node: NodeCore, dictionary: string[]) {
+function readBlobSection(node: NodeCore) {
     const blobs: Map<string, ArrayBuffer> = new Map();
-    for (const [idIndexNode, blobNode] of node.iteratePairs()) {
-        assertNumberInstance(idIndexNode, "Blob index should be a number");
-        assertBlobCoreInstance(blobNode, "Blob content should be of type blob");
-        const blobId = dictionary[idIndexNode];
-        assert(blobId !== undefined, 0x214 /* "blob id should be present" */);
-        blobs.set(blobId, blobNode.arrayBuffer);
+    for (let count = 0; count < node.length; ++count) {
+        const blob = node.getNode(count);
+        const records = getAndValidateNodeProps(blob, ["id", "data"]);
+        assertBlobCoreInstance(records.data, "data should be of BlobCore type");
```

Comment:
Do we have an issue tracking tagging assertBlobCoreInstance and similar APIs to remove strings from final bundles?

Reply:
Yes it is here: https://github.com/microsoft/FluidFramework/issues/7185